### PR TITLE
Fix some Tux animation glitches

### LINKED ISSUE #2533: Earth Tux Animation Bug
**SuperTux version:Last nightly for windows**
**System information:** *(details about your kernel, operating system, or distribution)*

##### Expected behavior
When taking an earth flower while pressing ACTION and DOWN, you should have the grow animation and THEN the roll animation
##### Actual behavior
It's making both
##### Steps to reproduce actual behavior
Take an earth flower while pressing ACTION and DOWN.
##### Additional debugging information
[https://youtu.be/NK-FQMNIi9M](url)

### PR DESCRIPTION
**Pull request by Brockengespenst**
* Wait until the growing animation is finished before changing to stone animation when smallTux gains an earth flower.
* Fix animation glitch from sliding to slidegrow
* Remain in crawling action when down is held and Tux stops sliding due to friction

Fixes #2533

### CODE REVIEW

**Comment by Brockengespenst** [comment]
I tried as best as possible to workaround this. Even though this might occur only very rare, it is worth having a look at it. The growing animation is now finished before turning to stone. Transition from growing to stone is not that smooth, to be honest.

Beside this, I noticed some animation glitches when small Tux slides and gains a grow powerup (the hight was adjusted for a small time which led Tux "fly" for a short time), which should be fixed as well.

I did several tests, but someone else (@Rusty-Box ?) should definitely do this as well and provide feedback.

File: src/object/player.cpp [hunk 1]
```diff
@@ -1588,11 +1593,11 @@ Player::handle_input()
...
 
   /* Duck or Standup! */
-  if ((m_controller->pressed(Control::DOWN) || ((m_duck || m_wants_buttjump) && m_controller->hold(Control::DOWN))) &&
-    !m_swimming && !m_sliding && !m_stone) {
+  if (((m_controller->pressed(Control::DOWN) && !m_growing && !m_stone) || ((m_duck || m_wants_buttjump) && m_controller->hold(Control::DOWN))) &&
+    !m_swimming && !m_sliding && !m_stone && !m_growing && !m_slidegrowing) {
     do_duck();
   }
-  else {
+  else if (!m_slidegrowing){
```
Comment (tobbi) [inline src/object/player.cpp:1600]: Add a space here.

**Review by tobbi — APPROVED** [review]
Didn't test this, but code looks good.

**Review by weluvgoatz — APPROVED** [review]

**Comment by Brockengespenst** [comment]
Looks like recent changes on master broke some of this. I need to rework this.

**Comment by Brockengespenst** [comment]
Ok, next try. Mentionend points in description should be included now.
@weluvgoatz : Would be great if you could check that I did not reintrocude #2838.

**Review by weluvgoatz — APPROVED** [review]

**Review by mrkubax10 — APPROVED** [review]

**Review by Rusty-Box — APPROVED** [review]

**Pull request merged**